File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        map<string,vector<string>> m;
        for(string s : strs) {
            string x = s;
            sort(x.begin(),x.end());
            m[x].push_back(s);
        }
        for(auto val : m) {
            ans.push_back(val.second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,ArrayList<String>> map = new HashMap<>();
        List<List<String>> list = new ArrayList<>();
        int n = strs.length;
        for(String s : strs) {
            char x[] = s.toCharArray();
            Arrays.sort(x);
            String a = "";
            for(char ch : x) a += ch;
            if(map.containsKey(a)) map.get(a).add(s);
            else {
                map.put(a,new ArrayList<>());
                map.get(a).add(s);
            }
        }
        for(ArrayList<String> z : map.values()) {
            list.add(z);
        }
        return list;
    }
}